test/metabase/test/util.clj

```diff
@@ -997,3 +997,9 @@
                  (macroexpand form))
       `(with-temp-file [])
       `(with-temp-file (+ 1 2)))))
+
+(defn dash->underscore
+  "Replaces dashes with underscores in `nm`. Intended for use in driver test data classes, to remap database, table,
+  etc. names in DBs that don't support having dashes in the identifier."
+  [nm]
+  (str/replace nm #"-" "_"))
```

Comment:
We already have `metabase.util/snake-key` that does this. But we don't even really need to be implementing that ourselves since [`camel-snake-kebab`](https://github.com/clj-commons/camel-snake-kebab) is a thing

Reply:
As discussed on Slack, that library is also converting to lowercase in addition to snake casing.  So we will stick with our own impl for now.